editor.planx.uk/src/@planx/components/FindProperty/Public/Map.tsx

```diff
@@ -0,0 +1,136 @@
+  teamSettings?: TeamSettings;
+  id?: string;
+  setAddress: React.Dispatch<React.SetStateAction<SiteAddress | undefined>>;
+  setPage: React.Dispatch<React.SetStateAction<"os-address" | "new-address">>;
+}
+
+export const DescriptionInput = styled(Box)(({ theme }) => ({
+  display: "flex",
+  paddingTop: theme.spacing(2),
+  paddingBottom: theme.spacing(2),
+}));
+
+export default function PlotNewAddress(props: PlotNewAddressProps) {
+  const [siteDescription, setSiteDescription] = useState<string | undefined>(
+    props.initialProposedAddress?.title ?? undefined
+  );
+  const [proposedAddress, setProposedAddress] = useState<
+    SiteAddress | undefined
+  >(props.initialProposedAddress ?? undefined);
+
+  const environment = useStore((state) => state.previewEnvironment);
+
+  useEffect(() => {
+    const geojsonChangeHandler = ({ detail: geojson }: any) => {
+      if (geojson["EPSG:3857"]?.features && geojson["EPSG:27700"]?.features) {
+        setProposedAddress({
+          // only a single point can be plotted, so get first feature in geojson "FeatureCollection" per projection
+          longitude: geojson["EPSG:3857"].features[0]?.geometry?.coordinates[0],
+          latitude: geojson["EPSG:3857"].features[0]?.geometry?.coordinates[1],
+          x: geojson["EPSG:27700"].features[0]?.geometry?.coordinates[0],
+          y: geojson["EPSG:27700"].features[0]?.geometry?.coordinates[1],
+          title: siteDescription || "",
+          source: "proposed",
+        });
+      } else {
+        // triggered if a user "clears" their point on the map
+        setProposedAddress(undefined);
+      }
+    };
+
+    const map: any = document.getElementById("plot-new-address-map");
+    map?.addEventListener("geojsonChange", geojsonChangeHandler);
+
+    return function cleanup() {
+      map?.removeEventListener("geojsonChange", geojsonChangeHandler);
+    };
+  }, [setProposedAddress]);
+
+  return (
+    <Card
+      handleSubmit={() => {
+        //@ts-ignore
+        props.setAddress({ ...proposedAddress, title: siteDescription });
+      }}
+      isValid={Boolean(proposedAddress) && Boolean(siteDescription)}
+    >
+      <QuestionHeader
+        title={props.title || DEFAULT_NEW_ADDRESS_TITLE}
+        description={props.description || ""}
+      />
+      <MapContainer environment={environment} interactive>
+        <p style={visuallyHidden}>
+          An interactive map centred on the local authority district, showing
+          the Ordnance Survey basemap. Click to place a point representing your
+          proposed site location.
+        </p>
+        {/* @ts-ignore */}
+        <my-map
+          id="plot-new-address-map"
+          data-testid="map-web-component"
+          zoom={14}
+          drawMode
+          drawType="Point"
+          resetControlImage="trash"
+          showScale
+          showNorthArrow
+          osVectorTilesApiKey={process.env.REACT_APP_ORDNANCE_SURVEY_KEY}
+        />
```

Comment:
Here's an interesting question: the starting viewport of this map is currently just the component default (London), but will need to reflect the "team" - so where do we want to store the center coordinate of each team?

Two ideas:
- Expose as fields in the Editor, fallback to component default if not provided. Pros: editors manage!; cons: this might be a more confusing content ask than normal (eg possibility of mixing up x/y, wrong projections, rounding, etc)
- Add to `teams.settings` in the database. Pros: we control data quality; cons: slow feedback loop if changes requested

Any strong opinions on which route? Any other ideas not on my list?

Reply:
Also, I'm totally expecting a piece of feedback here is going to be "can the map have search" rather than just zooming/panning interactions alone. 

In my mind, that's a v2.0 feature and the implementation would happen in the web component repo. I think the OS Places API `/find` endpoint would be a good fit for custom free-text search, returning a record with a lat/lng we could use to update viewport https://osdatahub.os.uk/docs/places/technicalSpecification

Reply:
:bulb: had a nice realization here: we can simply store a GeoJSON URL from Digital Land representing the boundary we want for a given team. 

This gives us lots of flexibility to decide whether that's the `local-authority-district` boundary for this team (probably the best fit most often) or something else if it's an edge case (eg mergers, etc). As much as I'd love a good reason to use PostGIS column data types and methods, fetching and benefitting from someone else's maintained boundary does the trick here :upside_down_face: 

Example for Braintree: `teams.settings.boundary = "https://www.planning.data.gov.uk/entity/8600093.geojson"`

I simply searched Digital Land's datasets and landed on this page https://www.planning.data.gov.uk/entity/8600093, then clicked "Download geojson" from the bottom map view, and added the URL to Hasura :heavy_check_mark: 

If `teams.settings.boundary` isn't set, we'll simply default to London.

Will add an issue to the onboarding scripts repo to capture this step there too :+1:

Reply:
Result is a nicely centered map (geojson boundary is light grey for now, could easily be made more bold in future) 
![Screenshot from 2023-02-07 18-47-32](https://user-images.githubusercontent.com/5132349/217324301-ab3ece2e-8d3b-4fba-a134-49ab319e3b17.png)

Reply:
This is a great solution - best of both worlds.
 - No maintenance (unless broken by Digital Land - unlikely!)
 - Boundary is richer and more useful than centroid
 - Can be handled pretty easily through onboarding, much more easily than pasting a geojson etc

Nice!